src/ophys_etl/modules/suite2p_registration/suite2p_utils.py

```diff
@@ -320,17 +331,59 @@ def optimize_motion_parameters(initial_frames: np.ndarray,
     return best_results
 
 
+def load_representitive_sub_frames(h5py_name,
+                                   h5py_key,
+                                   trim_frames_start: int = 0,
+                                   trim_frames_end: int = 0,
+                                   n_batches: int = 20,
+                                   batch_size: int = 500):
+    """Load a subset of frames spanning the full movie.
+
+    Parameters
+    ----------
+    h5py_name : str
+        Name of h5 file to load frames from.
+    h5py_key : str
+        Name of the h5 dataset containing the movie.
+    trim_frames_start : int, optional
+        Number of frames to disregard from the start of the movie. Default 0.
+    trim_frames_start : int, optional
+        Number of frames to disregard from the end of the movie. Default 0.
+    n_batches : int
+        Number of batches to load. Total returned size is
+        n_batches * batch_size.
+    batch_size : int, optional
+        Number of frames to process at once. Total returned size is
+        n_batches * batch_size.
+
+    Returns
+    -------
+    """
+    output_frames = []
+    frame_percents = np.arange(0, 1, 1 / n_batches)
+    with h5py.File(h5py_name) as h5_file:
+        dataset = h5_file[h5py_key]
+        total_frames = dataset.shape[0] - trim_frames_start - trim_frames_end
+        if total_frames < n_batches * batch_size:
+            return dataset[:]
+        for percent_start in frame_percents:
+            frame_start = int(percent_start * total_frames + trim_frames_start)
+            output_frames.append(dataset[frame_start:frame_start + batch_size])
+    return np.concatenate(output_frames)
+
+
 def create_ave_image(ref_image: np.ndarray,
+                     input_frames: np.ndarray,
                      suite2p_args: dict,
-                     trim_frames_start: int = 0,
-                     trim_frames_end: int = 0,
                      batch_size: int = 500) -> dict:
     """Run suite2p image motion correction over a full movie.
 
     Parameters
     ----------
     ref_image : numpy.ndarray, (N, M)
         Reference image to correlate with movie frames.
+    input_frames : numpy.ndarray, (L, N, M)
+        Frames to motion correcdt and compute average image/acutance of.
     suite2p_args : dict
         Dictionary of suite2p args containing:
```

Comment:
Need to update the list of required args. I don't think you use h5py or h5py_key (...right? Not sure what's under the hood in register_image)

Reply:
Those are still required args in the suite2p call.